src/pages/Index/Section/Personal/index.js

```diff
@@ -36,16 +63,30 @@ const Header = ({ tabState }) => (
 )
 
 const Personal = () => {
-  const tabState = useState(loadTab)
-  const activeTab = tabState[0]
+  const [activeTab, setTab] = useState(loadTab)
   const Content = TabTypeComponent[activeTab]
 
+  const history = useHistory()
+
   useEffect(() => saveTab(activeTab), [activeTab])
```

Comment:
This effect is called on the tab change, therefore hash update should be here. For example:
```js
const hash = TabHash[activeTab] || '';
history.replace(window.location.pathname + hash);
```